### DISCUSSION TITLE
Can I disable dns with `podman kube`?

### DISCUSSION THREAD

**Discussion by Epokhe** [containers/podman#19167] [Q&A]
`podman generate kube` generates `dnsConfig` entry for `--dns`. But it doesn't generate anything for `--dns=none`.  

`none` allows you to disable dns according to https://docs.podman.io/en/latest/markdown/podman-run.1.html#dns-ipaddr

The feature PR is here: https://github.com/containers/podman/pull/9150. I haven't bothered to understand the code, but it doesn't include a test for `none` value. I am not sure if `none` case is supported or not.

**Comment by rhatdan** [comment]
Just fooling around with this in `podman kube generate` it looks like the kube.yaml does not do anything with --dns=none.  Should it?  Does kubernetes support the idea of disabling dns?
```
$ podman create --dns=none --name mykube alpine echo hi
be70a441a94fa8d632988a8cb82b0163d53916313517d899d38be286f1f8b28b
$ podman kube generate be70a441a94fa8d632988a8cb82b0163d53916313517d899d38be286f1f8b28b
# Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-4.6.0-dev
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2023-07-10T16:21:55Z"
  labels:
    app: mykube-pod
  name: mykube-pod
spec:
  containers:
  - command:
    - echo
    - hi
    env:
    - name: QEMU_CPU
      value: max
    image: docker.io/library/alpine:latest
    name: mykube
```

```
$ podman create --dns=4.2.2.2 --replace --name mykube alpine echo hi
mykube
975388a397ff176224bfcbc56d741cac10bff4f9bdea66279858d307dc7c5b74
$ podman kube generate mykube
# Save the output of this file and use kubectl create -f to import
# it into Kubernetes.
#
# Created with podman-4.6.0-dev
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2023-07-10T16:22:24Z"
  labels:
    app: mykube-pod
  name: mykube-pod
spec:
  containers:
  - command:
    - echo
    - hi
    env:
    - name: QEMU_CPU
      value: max
    image: docker.io/library/alpine:latest
    name: mykube
  dnsConfig:
    nameservers:
    - 4.2.2.2
```

**Reply by rhatdan** [reply]
Am I correct in assuming the first example should have been

 ` dnsPolicy: "None"` in the containers: section?

**Reply by rhatdan** [reply]
@umohnani8 @ygalblum PTAL

**Reply by umohnani8** [reply]
In k8s, when `dnsPolicy: None`, the `dnsConfig` field needs to be specified as well https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config.
Is `--dns=none` in podman the same as `dnsPolicy:none` in k8s? (it doesn't look like it) If so, we should be generating the `dnsPolicy: none` in the kube yaml but also need to generate a `dnsConfig` with it, which I am not sure what the podman equivalent is.

**Reply by rhatdan** [reply]
--dns=none, will not generate a resolv.conf, so that resolv.conf inside of the container image, or the lack of one will be used.
It does not look like this data is recorded inside of the containers database, other then the fact that a resolv.conf file does not get created in the container data.  If K8s does not have similar functionality, I do know know what we need to do.
